server/client.go

```diff
@@ -1590,6 +1599,8 @@ func (c *client) flushOutbound() bool {
 		}
 		if err != nil {
 			c.Errorf("Error compressing data: %v", err)
+			// We need to grab the lock now before marking as closed and exiting
+			c.mu.Lock()
```

Comment:
That was missing since we are in a location where the lock was released but we are failing and returning before acquiring the lock after the TCP write. The caller is expected that the function returns with lock held.